packages/eslint-plugin/src/rules/no-deprecated-components/createNoDeprecatedComponentsRule.ts

```diff
@@ -112,10 +129,34 @@ export function createNoDeprecatedComponentsRule(
                             messageId: "migration",
                             node,
                         });
+                    } else if (isOpeningElement(node.parent)) {
+                        // check <DeprecatedComponent withDeprecatedProp={...}> syntax
+                        const deprecatedProp = node.parent.attributes.find(
+                            attribute =>
+                                attribute.type === TSESTree.AST_NODE_TYPES.JSXAttribute &&
+                                attribute.name.type === TSESTree.AST_NODE_TYPES.JSXIdentifier &&
+                                deprecatedToNewComponentMapping[`${node.name}.${attribute.name.name}`] != null,
+                        );
+                        if (deprecatedProp !== undefined) {
+                            const deprecatedComponentKey = Object.keys(deprecatedToNewComponentMapping).find(
+                                key => key.split(".")[0] === node.name,
```

Comment:
it is technically possible to have both `MenuItem` and `MenuItem.popoverProps` as keys, in which case this would return the wrong key, right?
```suggestion
                                key => key.indexOf(".") >= 0 && key.split(".")[0] === node.name,
```

Reply:
Yeah, I think it would fail in that case. The rule doesn't gracefully handle bad configuration right now, but I figure it's ok to be a little simplistic here since the configuration is not exposed to consumers -- it only lives inside the rule implementation in this package.

Reply:
fixed it up a little to guard against the edge case you mentioned